cmd/csi-cluster-driver-registrar/k8s_register.go

```diff
@@ -41,24 +42,25 @@ func kubernetesRegister(
 		os.Exit(1)
 	}
 
-	// Set up goroutine to cleanup (aka deregister) on termination.
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt)
-	go cleanup(c, clientset, csiDriver)
-
-	// Run forever
+	// Run until killed and in the meantime, regularly ensure that the CSIDriverInfo is set.
+	c := make(chan os.Signal, 3)
+	t := time.Tick(sleepDuration)
+	signal.Notify(c,
+		syscall.SIGQUIT,
+		syscall.SIGTERM,
+		syscall.SIGINT)
 	for {
 		verifyAndAddCSIDriverInfo(clientset, csiDriver)
-		time.Sleep(sleepDuration)
+		select {
+		case s := <-c:
+			klog.V(1).Infof("signal %q caught, removing CSIDriver object", s)
+			verifyAndDeleteCSIDriverInfo(clientset, csiDriver)
+			return
+		case <-t:
```

Comment:
I would suggest changing to the following that way the code can avoid the possibility of leaking ticker.  Plus no matter what causes an exit, verifyAndDeleteCSIDriver will be called. 

```go
ticker := time.NewTicker(time.Second)
defer func(){
    verifyAndDeleteCSIDriverInfo(clientset, csiDriver)
    ticker.Stop()
}()
for {
        verifyAndAddCSIDriverInfo(clientset, csiDriver)
	select {
	case <-c:
            klog.V(1).Infof("signal %q caught, removing CSIDriver object", s)
            return
	case <-ticker.C:
	}
}

```

Reply:
Good suggestion. Changed.